### DISCUSSION TITLE
Consider migrating to ES2018 or later to avoid typescript generating emulation of forawait loop

### DISCUSSION THREAD

**Discussion by constantind** [enisdenjo/graphql-ws#551] [General]
Hi @enisdenjo 

Due to es2017 https://github.com/enisdenjo/graphql-ws/blob/master/tsconfig.json#L4 typescript generates emulation with try/catch/finally in the final js file for this code:  https://github.com/enisdenjo/graphql-ws/blob/master/src/server.ts#L819
Instead this should be expected to utilize the native for await as is: 
for await (const result of operationResult)

Currently with npm install unable to force es2022, any possible workarounds or add support for es2018?

Thank you

**Comment by enisdenjo** [comment]
Just a short update, we're on a brink of a next major version. Mainly because of #553. I reckon I'll pack a Node version bump too requiring at least the LTS at the time being (v20).

**Reply by constantind** [reply]
Thank you, Regarding 553 we have same issue with express and node18, which i think it is nodejs issue rather than websocket issue or rather an issue with streams at the end the tcp socket is also a stream but it would be better to be an option so other sockets work, Another problem could be for instance, one could just add the winston-transport and get readable/writable streams overwritten instead of node defaults, then add workaround elsewhere.
